<launch>
  <arg name="INPUT_IMAGE"/>
  <arg name="IMAGE_TRANSPORT" default="raw"/>
  <arg name="model_file" default="$(find coral_usb)/models/mobilenet_ssd_v2_coco_quant_postprocess_edgetpu.tflite"/>
  <arg name="label_file" default="$(find coral_usb)/models/coco_labels.txt"/>
  <arg name="device_id" default="0" />
  <arg name="n_split" default="3" />

  <node name="edgetpu_panorama_object_detector"
        pkg="coral_usb" type="edgetpu_panorama_object_detector.py"
        output="screen" respawn="true">
    <remap from="~input" to="$(arg INPUT_IMAGE)" />
    <rosparam subst_value="true" >
      model_file: $(arg model_file)
      label_file: $(arg label_file)
      image_transport: $(arg IMAGE_TRANSPORT)
      device_id: $(arg device_id)
      n_split: $(arg n_split)
    </rosparam>
  </node>
</launch>
